src/obu.rs

```diff
@@ -355,11 +355,14 @@ fn parse_seq_hdr(
     let trc;
     let mtrx;
     if color_description_present != 0 {
-        pri = Rav1dColorPrimaries(gb.get_bits(8) as u8);
+        pri = gb
```

Comment:
Nit: I'd spell this as

```rust
pri = match Rav1dColorPrimaries::try_from(gb.get_bits(8)) {
     Ok(pri) => pri,
     _ => Rav1dColorPrimaries::Unspecified,
};
```

Reply:
Why though? That just seems more wordy and non left-to-right logic that's harder to follow.  `.unwrap_or` is pretty idiomatic.

Reply:
I think it is easier to follow. IMO your version also relies on non left-to-right logic in that you have to look at the `unwrap_or()` clause to get the inferred type for `try_into`.